Python/pylifecycle.c

```diff
@@ -1752,13 +1752,16 @@ finalize_interp_types(PyInterpreterState *interp)
     _PyUnicode_ClearInterned(interp);
 
     _PyDict_Fini(interp);
-    _PyList_Fini(interp);
     _PyTuple_Fini(interp);
 
     _PySlice_Fini(interp);
 
     _PyUnicode_Fini(interp);
     _PyFloat_Fini(interp);
+
+    _PyFreeListState *state = _PyFreeListState_GET();
```

Comment:
@colesbury
Since both free-threading and GIL interpreter should finalize their own free lists (per thread or per interpreter),
It should be managed in a unified and seamless way.
Please check that this code is correct :)

Reply:
Okay, so on the free-threading build it could be redundant call because it will call `_PyXXX_ClearFreeList ` again,
but since it's already called, there will be no loop iteration and the only effect of this calling will be ensuring that the freelist is finalized.
https://github.com/python/cpython/commit/bcb198385dee469d630a184182df9dc1463e2c47

Reply:
Ah forget about above comments, it said 'Destroy the state of all threads of the interpreter, except of the current thread.' so this calling for freethreading build is actually needed.